src/storm/storage/expressions/Expression.cpp

```diff
@@ -300,9 +300,13 @@ Expression operator/(Expression const& first, Expression const& second) {
 
 Expression operator%(Expression const& first, Expression const& second) {
     assertSameManager(first.getBaseExpression(), second.getBaseExpression());
-    return Expression(std::shared_ptr<BaseExpression>(new BinaryNumericalFunctionExpression(
+    // First we compute the remainder
+    Expression remainder(std::shared_ptr<BaseExpression>(new BinaryNumericalFunctionExpression(
         first.getBaseExpression().getManager(), first.getType().modulo(second.getType()), first.getBaseExpressionPointer(), second.getBaseExpressionPointer(),
         BinaryNumericalFunctionExpression::OperatorType::Modulo)));
+
+    // Deal with negative `first` in accordance with Prism
```

Comment:
```suggestion
    // Deal with negative `first` in accordance with Prism and Jani
```
The Jani spec says:
> Unlike e.g. in most implementations of C, this operation computes the remainder of the Euclidean division: the result is always positive and x % 0 is division by zero.

As far as I can see, this is what we are doing now with this fix, right?

Reply:
Yes, except for the `x % 0` part. For normal integers that is going to give some divide/modulo zero error.